.github/workflows/build_docs.yml

```diff
@@ -0,0 +1,76 @@
+name: Build and Deploy Documentation
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'docs/**'
+      - '.github/workflows/build_docs.yml'
+  pull_request:
+    paths:
+      - 'docs/**'
+      - '.github/workflows/build_docs.yml'
+  workflow_dispatch:
+
+jobs:
+  build_docs:
+    name: Build Mesh Adaptation Documentation
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/mesh-adaptation/firedrake-um2n:latest
+      options: --user root
+      volumes:
+        - ${{ github.workspace }}:/home/firedrake/output
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Animate and Goalie
+        run: |
+          . /home/firedrake/firedrake/bin/activate
+          cd /home/firedrake/firedrake/src/
+          git clone https://github.com/mesh-adaptation/animate.git
+          git clone https://github.com/mesh-adaptation/goalie.git
+          pip install -e animate
+          pip install -e goalie
+        working-directory: /home/firedrake/output/
+        shell: bash
+
+      - name: Install Sphinx
```

Comment:
"and other dependencies"

Reply:
Done in a04dccd9a824cab40e57fcef0fdfd6e9d810d3c4